### PR TITLE
[jenkins] do not load unnecessary code in case of webdav

### PR DESCRIPTION
**Pull request by DeepDiver1975**
refs https://github.com/owncloud/core/pull/19144

changing from "protected static" to "protected"
as suggested by @nickvergessen
https://github.com/owncloud/core/pull/19114#discussion_r39719851

moving initTemplate() into template constr.

reduce to move initTemplate only

cleanup spaces

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
failing oracle is unrelated

**Comment by DeepDiver1975** [comment]
Please review @icewind1991 @blizzz @schiesbn @PVince81 THX

File: lib/private/template.php [hunk 1]
```diff
@@ -56,8 +56,13 @@ class OC_Template extends \OC\Template\Base {
 	 * according layout. For now, renderas can be set to "guest", "user" or
 	 * "admin".
 	 */
+	
+	protected static $initTemplateEngineFirstrun = true;
+	
 	public function __construct( $app, $name, $renderas = "", $registerCall = true ) {
 		// Read the selected theme from the config file
+		$this->initTemplateEngine();
```
Comment (icewind1991) [inline lib/private/template.php]: should that be `self::...`

File: lib/private/template.php [hunk 2]
```diff
@@ -76,6 +81,78 @@ public function __construct( $app, $name, $renderas = "", $registerCall = true )
 		parent::__construct($template, $requesttoken, $l10n, $themeDefaults);
 	}
 
+	public static function initTemplateEngine() {
+		if (self::$initTemplateEngineFirstrun){
+			
+			//apps that started before the template initialization can load their own scripts
+			//so to make sure this scripts here are loaded first we use OC_Util::addScript() with $prepend=true
```
Comment (phil-davis) [inline lib/private/template.php]: s/this scripts/the scripts

File: lib/private/template.php [hunk 3]
```diff
@@ -76,6 +81,78 @@ public function __construct( $app, $name, $renderas = "", $registerCall = true )
...
+
+			OC_Util::addStyle ( "styles" );
+			OC_Util::addStyle ( "header" );
+			OC_Util::addStyle ( "mobile" );
+			OC_Util::addStyle ( "icons" );
+			OC_Util::addStyle ( "fonts" );
+			OC_Util::addStyle ( "apps" );
+			OC_Util::addStyle ( "fixes" );
+			OC_Util::addStyle ( "multiselect" );
+			OC_Util::addVendorStyle ( 'jquery-ui/themes/base/jquery-ui');
```
Comment (phil-davis) [inline lib/private/template.php]: How much does ownCloud fuss about code style? At the end of this line a space is missing before the closing );
My brain felt unbalanced seeing an opening space, and the similar spacing of the other lines, and then the missing space here :)

**Comment by DeepDiver1975** [comment]
> Please review @icewind1991 @blizzz @schiesbn @PVince81 THX

once again please review @icewind1991 @blizzz @schiesbn @PVince81 THX

**Comment by PVince81** [comment]
Update still works, webdav still works, web ui upload still work :+1:

**Comment by PVince81** [comment]
Hmm wait, it seems to break some styles

**Comment by PVince81** [comment]
:-1:

master:
upload-menu-style-master

this branch (rebased locally on top of master):
upload-menu-style-branch

I cleared the browser cache when testing this.

**Comment by DeepDiver1975** [comment]
@individual-it can I ask you to have a look at the broken style? Maybe a different load order of sytles/js ??? THX

**Comment by individual-it** [comment]
changing the order of loading the CSS should have fixed the issue.

**Comment by DeepDiver1975** [comment]
@PVince81 mind retesting? THX

**Comment by icewind1991** [comment]
The issue @PVince81 found is gone for me after pulling, can't find any further style issues, code looks good :+1:

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by MorrisJobke** [comment]
This PR bricks l10n. After this everything is english even if you set up a different language ...I will try to fix this

**Comment by DeepDiver1975** [comment]
> This PR bricks l10n.

wow - how that?

**Comment by MorrisJobke** [comment]
I have no idea ...

**Comment by MorrisJobke** [comment]
> I have no idea ...

git bisect lied to me :sob: I will continue my investigation.

**Comment by individual-it** [comment]
l10n works fine for me
i10n

**Comment by MorrisJobke** [comment]
> l10n works fine for me

Yes. Sorry for the noise. This was unrelated.

**Pull request merged**